tests/test_scanner_headless.py

```
@@ -425,6 +425,65 @@ async def fake_render(url):
+
+        def fake_detect(html, headers, cookies):
+            # The rendered DOM exposes a third system the thin static body hid.
+            if "RENDERED_DOM" in (html or ""):
+                return [_sys("Cloudflare"), _sys("Google Analytics"), _sys("Akeneo")]
+            return [_sys("Cloudflare"), _sys("Google Analytics")]  # only 2 statically
+
+        monkeypatch.setattr(scanner, "_detect_systems_generic", fake_detect)
+
+        class _Resp:
+            text = "<html><body>thin storefront</body></html>"
+            headers: dict = {}
```

Ruff (0.15.15)

[warning] Mutable default value for class attribute

(RUF012)